python/pyarrow/tests/test_pandas.py

```diff
@@ -95,9 +96,13 @@ def _check_pandas_roundtrip(df, expected=None, use_threads=False,
     if expected is None:
         expected = df
 
-    tm.assert_frame_equal(result, expected, check_dtype=check_dtype,
-                          check_index_type=('equiv' if preserve_index
-                                            else False))
+    # pandas.testing generates a
+    # DeprecationWarning: elementwise comparison failed
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", DeprecationWarning)
+        tm.assert_frame_equal(result, expected, check_dtype=check_dtype,
+                              check_index_type=('equiv' if preserve_index
+                                                else False))
```

Comment:
Could we be more specific in which warning we are catching? (so are not catching any future deprecation warning, but only specifically the current numpy one)

(for example with `filterwarnings()` instead of `simplefilter` where we can also specify the module / message, similarly to the `pytest.mark.filterwarnings`. Or just use that mark)

Reply:
Good point. Done in https://github.com/apache/arrow/pull/14729/commits/ee97c6bdc475b6e8e78c827cfea278272189f5c8